xarray/core/accessors.py

```diff
@@ -58,6 +58,49 @@ def _get_date_field(values, name, dtype):
         return _access_through_series(values, name)
 
 
+def _round_series(values, name, freq):
+    """Coerce an array of datetime-like values to a pandas Series and
+    apply requested rounding
+    """
+    values_as_series = pd.Series(values.ravel())
+    if name == 'floor':
+        field_values = values_as_series.dt.floor(freq=freq).values
+    if name == 'ceil':
+        field_values = values_as_series.dt.ceil(freq=freq).values
+    if name == 'round':
+        field_values = values_as_series.dt.round(freq=freq).values
+
+    return field_values.reshape(values.shape)
+
+
+def _round_field(values, name, freq, dtype):
+    """Indirectly access pandas rounding functions by wrapping data
+    as a Series and calling through `.dt` attribute.
+
+    Parameters
+    ----------
+    values : np.ndarray or dask.array-like
+        Array-like container of datetime-like values
+    name : str (ceil, floor, round)
+        Name of rounding function
+    freq : a freq string indicating the rounding resolution
+    dtype : dtype-like
+        dtype for output date field values
+
+    Returns
+    -------
+    rounded timestamps : same type as values
+        Array-like of datetime fields accessed for each element in values
+
+    """
+    if isinstance(values, dask_array_type):
+        from dask.array import map_blocks
+        return map_blocks(_round_series,
+                          values, name, freq=freq, dtype=dtype)
+    else:
+        return _round_series(values, name, freq)
```

Comment:
At some point, we should probably consolidate this function with the similar looking one above. But I'm OK with this for now.

Reply:
Ya, I have no experience with dask so I just copied the previous one.